### ISSUE TITLE
Considering Updating to GPT-4 from GPT-3

### ISSUE THREAD

**Issue by ash1eygrace** [ash1eygrace/ai-content#44]
Just received the invitation to access the new GPT-4 models in Beta via the OpenAI API, so it's now testing time. Specifically, explore `gpt-4`, `gpt-4-0314`, and `gpt-4-32k` since it seems that the `gpt-4-32k-0314` will not receive updates, and support will end on June 14th, 2023.

From briefly skimming the details, it seems GPT-4 offers improvements over the GPT-3.5 models, with greater accuracy, broader general knowledge, and advanced reasoning capabilities. The latest GPT-4 models support a maximum of 8,192 tokens, and their training data is up to September 2021.

That said, the implementation from GPT-3 to GPT-4 will mean improved performance in complex reasoning tasks. GPT-4 is optimized for chat but works well for traditional completions tasks. 

The transition will likely involve changes to the API calls and possibly some adjustments to the application logic.
To proceed with the update the following will need to happen:
1. Assess the potential benefits of GPT-4 for the app's current specific use case(s), aside from the Chat Integration.
2. Review the API documentation and determine what adjustments to the API calls need to be complete to utilize the GPT-4 models.
3. Test the updated implementation and compare the results against the current GPT-3-based system for the existing generator components (may need to try different prompt variations in the existing generator data).
5. Monitor usage and cost implications, as GPT-4 models may have different pricing compared to GPT-3.5 models. (I've never used over $0.67 a month, but still lol) 
6. Update and create any relevant documentation. 
7. Once the transition is complete, evaluate the potential to fine-tune the GPT-4 models for further customization and optimization for the app's existing use case(s).
